dsgrid/config/dataset_schema_handler_standard.py

```diff
@@ -0,0 +1,187 @@
+    DatasetConfig,
+    check_load_data_filename,
+    check_load_data_lookup_filename,
+)
+from dsgrid.utils.spark import read_dataframe, get_unique_values
+from dsgrid.utils.timing import timer_stats_collector, Timer
+from dsgrid.config.dataset_schema_handler_base import DatasetSchemaHandlerBase
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.exceptions import (
+    DSGInvalidDataset,
+    DSGInvalidDimension,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class StandardDatasetSchemaHandler(DatasetSchemaHandlerBase):
+    """ define interface/required behaviors for STANDARD dataset schema """
+
+    def __init__(self, config):
+        self._config = config
+
+    def get_pivot_dimension_columns(self):
+        """ get cols for the dimension that is pivoted in load_data. """
+        dim_type = self._config.model.data_schema.load_data_column_dimension
+        return self._config.get_dimension(dim_type).get_unique_ids()
+
+    def check_consistency(self):
+        path = Path(self._config.model.path)
+        load_data_df = read_dataframe(check_load_data_filename(path))
+        load_data_lookup = read_dataframe(check_load_data_lookup_filename(path), cache=True)
+        load_data_lookup = self._config.add_trivial_dimensions(load_data_lookup)
+        with Timer(timer_stats_collector, "check_lookup_data_consistency"):
+            self._check_lookup_data_consistency(self._config, load_data_lookup)
+        with Timer(timer_stats_collector, "check_dataset_time_consistency"):
+            self._check_dataset_time_consistency(self._config, load_data_df)
+        with Timer(timer_stats_collector, "check_dataset_internal_consistency"):
+            self._check_dataset_internal_consistency(self._config, load_data_df, load_data_lookup)
+
+    def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup):
+        found_id = False
+        dimension_types = []
+        for col in load_data_lookup.columns:
+            if col == "id":
+                found_id = True
+                continue
+            try:
+                dimension_types.append(DimensionType(col))
+            except ValueError:
+                raise DSGInvalidDimension(f"load_data_lookup column={col} is not a dimension type")
+
+        if not found_id:
+            raise DSGInvalidDataset("load_data_lookup does not include an 'id' column")
+
+        load_data_dimensions = (
+            DimensionType.TIME,
+            config.model.data_schema.load_data_column_dimension,
+        )
+        expected_dimensions = [d for d in DimensionType if d not in load_data_dimensions]
+        if len(dimension_types) != len(expected_dimensions):
+            raise DSGInvalidDataset(
+                f"load_data_lookup does not have the correct number of dimensions specified between trivial and non-trivial dimensions."
+            )
+
+        missing_dimensions = set(expected_dimensions).difference(dimension_types)
```

Comment:
This has the same no-op problem as in one_table_schema, but in this case I think you maybe could have len(dimension_types) > len(expected_dimensions)